src/test/java/uk/gov/ida/eventemitter/AmazonSqsClientIntegrationTest.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.ida.eventemitter;
+
+import cloud.localstack.LocalstackTestRunner;
+import cloud.localstack.TestUtils;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.AmazonSQSException;
+import com.amazonaws.services.sqs.model.CreateQueueRequest;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@RunWith(LocalstackTestRunner.class)
+public class AmazonSqsClientIntegrationTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private static final UUID ID = UUID.randomUUID();
+    private static final DateTime TIMESTAMP = DateTime.now(DateTimeZone.UTC);
+    private static final String EVENT_TYPE = "Error Event";
+    private static final String ENCRYPTED_EVENT = "encryptedEvent";
+    private static final String QUEUE_NAME = "queueName";
+
+    private static AmazonSQS sqs;
+    private static AmazonSqsClient sqsClient;
+    private static String queueUrl;
+    private static Event event;
+
+    @BeforeClass
+    public static void setUp() {
+        final Map<String, String> details = new HashMap<>();
+        details.put("type", "network error");
+        event = new TestEvent(ID, TIMESTAMP, EVENT_TYPE, details);
+
+        sqs = TestUtils.getClientSQS();
+        sqs.createQueue(new CreateQueueRequest(QUEUE_NAME));
+        queueUrl = sqs.getQueueUrl(QUEUE_NAME).getQueueUrl();
+        sqsClient = new AmazonSqsClient(sqs, queueUrl);
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        sqs.deleteQueue(queueUrl);
+    }
+
+    @Test
+    public void shouldSendEventToSqs() {
```

Comment:
What do we want to get out of this test?

We already have a test that confirms that the lib works end to end (ie that we can send things to SQS), and we also have a unit test to confirm that we can use the AmazonSQS correctly.

What extra are we hoping to get from having this set of tests also?

Reply:
We want to have a separate integration test that test AmazonSqsClient only (i.e.this integration test does not need EventEmitter and it also does not need EventEmitterModule too) so that this test can tell us that AmazonSqsClient works properly or not.

Reply:
Why do we need a separate test though?

It feels like we are integration testing a very small amount of code that is already covered by our existing tests.

For example, I don't really see how this is going to catch something that wouldn't already be caught elsewhere

Reply:
AmazonSqsClientIntegrationTest has a test which can test AmazonSqsClient to ensure that it throws an exception after sending an event message to SQS where queue does not exist. This test is for AmazonSqsClient only. EventEmitterIntegrationTest does not do that and instead EventEmitter catches the exception thrown by AmazonSqsClient and logs it.

Reply:
Feels a lot like testing a third party library to me. I'd personally stick to testing the desired behaviour. 

As usual though, if I'm outvoted by the rest of the team then that's fine!

Reply:
It is not testing a third party library. It is for testing the class we created specifically AmazonSqsClient class. Unit testing this class only does not assure us that it will work in a real world.

Last week, Vanessa wanted me to add integration tests to test individual components rather than the whole EventEmitter. That is why I added this integration test to test the direct interaction between AmazonSqsClient and Amazon SQS.